### PR TITLE
Make ErrorInfo contructor public

### LINKED ISSUE #4980: Reporting causes NewPipe to crash
### Checklist

- [x] I am using the latest version - 0.20.4 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug

1. Go to any video where the comments fail to load.
(Or repeatedly switch videos quickly, you'll certainly get a video where comments don't load)
2. A little pop-up says "Sorry, something went wrong. <b>Report</b>."
3. Clicking Report causes the app to crash.

Could reproduce almost 70% of the time.

### Actual behaviour

NewPipe crashes without any crash report and goes into a loop of showing the splash screen and crashing.

### Expected behavior

I haven't actually used that report button before encountering this issue.
Expected behaviour: I guess app shouldn't crash

### Screenshots/Screen recordings

Screenshot_20201123-190704
This report button at bottom right.

Video in the screenshot: 
https://www.youtube.com/watch?v=5GSeWdjyr1c
### Logs

<details>
<summary>
<b>Logcat</b>
</summary>
<p>

```
11-23 19:03:42.791 I/Timeline(32688): Timeline: Activity_launch_request id:org.schabi.newpipe time:325274613
11-23 19:03:42.801 E/Parcel  (2906): Class not found when unmarshalling: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:42.801 E/Parcel  (2906): java.lang.ClassNotFoundException: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:42.801 E/Parcel  (2906): Caused by: java.lang.ClassNotFoundException: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:42.811 V/WindowManager(2906): addAppToken: AppWindowToken{24ddbd7 token=Token{e5cbf56 ActivityRecord{3d91571 u0 org.schabi.newpipe/.report.ErrorActivity t1710}}} to stack=1 task=1710 at 1
11-23 19:03:42.841 D/GameManagerService(2906): NotifyRunnable. pkg: org.schabi.newpipe, type: 4, isMinimized: false, isTunableApp: false
11-23 19:03:42.846 V/MARsPolicyManager(2906): updatePackagesScore PackageInfo name -- org.schabi.newpipe
11-23 19:03:42.966 E/ACRA    (32688): ACRA caught a IllegalAccessError for org.schabi.newpipe
11-23 19:03:42.966 E/ACRA    (32688): java.lang.IllegalAccessError: Method 'void org.schabi.newpipe.report.ErrorInfo.<init>(org.schabi.newpipe.report.UserAction, java.lang.String, java.lang.String, int)' is inaccessible to class 'org.schabi.newpipe.report.ErrorInfo$Creator' (declaration of 'org.schabi.newpipe.report.ErrorInfo$Creator' appears in /data/app/org.schabi.newpipe-1/base.apk)
11-23 19:03:42.966 E/ACRA    (32688): 	at org.schabi.newpipe.report.ErrorInfo$Creator.createFromParcel(Unknown Source)
11-23 19:03:42.966 E/ACRA    (32688): 	at org.schabi.newpipe.report.ErrorActivity.onCreate(ErrorActivity.java:209)
11-23 19:03:43.401 I/WindowManager(2906): Screenshot max retries 4 of Token{e5cbf56 ActivityRecord{3d91571 u0 org.schabi.newpipe/.report.ErrorActivity t1710 f}} appWin=Window{ef41a93 u0 d0 p32688 org.schabi.newpipe/org.schabi.newpipe.MainActivity} drawState=4
11-23 19:03:43.586 I/WindowState(2906): WIN DEATH: Window{ef41a93 u0 d0 p32688 org.schabi.newpipe/org.schabi.newpipe.MainActivity}
11-23 19:03:43.586 I/ActivityManager(2906): Process org.schabi.newpipe (pid 32688)(adj 0) has died(109,242)
11-23 19:03:43.591 D/ActivityManager(2906): removeProcessNameLocked mProcessNames.remove pid=32688 ,hash=23487259 ,name=org.schabi.newpipe
11-23 19:03:43.861 D/PackageManager(2906): getComponentMetadataForIconTray : org.schabi.newpipe.MainActivity does not exist in mServices
11-23 19:03:43.861 D/PackageManager(2906): getComponentMetadataForIconTray : org.schabi.newpipe.MainActivity does not exist in mProviders
11-23 19:03:43.861 D/PackageManager(2906): getComponentMetadataForIconTray : org.schabi.newpipe.MainActivity does not exist in mReceivers
11-23 19:03:43.866 I/ApplicationPackageManager(3274): load=org.schabi.newpipe, bg=96-96, dr=96-96
11-23 19:03:44.476 D/MountService(2906): getExternalStorageMountMode : final mountMode=3, uid : 10753, packageName : org.schabi.newpipe
11-23 19:03:44.491 I/ActivityManager(2906): Start proc 333:org.schabi.newpipe:acra/u0a753 for service org.schabi.newpipe/org.acra.sender.JobSenderService
11-23 19:03:44.491 W/SELinux (333): SELinux: seapp_context_lookup: seinfo=default, level=s0:c512,c768, pkgname=org.schabi.newpipe:acra 
11-23 19:03:44.551 W/System  (333): ClassLoader referenced unknown path: /data/app/org.schabi.newpipe-1/lib/arm
11-23 19:03:45.006 I/ACRA    (333): Sending report /data/user/0/org.schabi.newpipe/app_ACRA-approved/2020-11-23T19:03:43.036+05:30.stacktrace
11-23 19:03:45.016 I/Timeline(333): Timeline: Activity_launch_request id:org.schabi.newpipe time:325276839
11-23 19:03:45.021 E/Parcel  (2906): Class not found when unmarshalling: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:45.021 E/Parcel  (2906): java.lang.ClassNotFoundException: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:45.021 E/Parcel  (2906): Caused by: java.lang.ClassNotFoundException: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:45.026 I/ActivityManager(2906): isWidgetUsingPkg : org.schabi.newpipe no widget is using.
11-23 19:03:45.031 V/WindowManager(2906): addAppToken: AppWindowToken{ae768c7 token=Token{c1aa06 ActivityRecord{2493ce1 u0 org.schabi.newpipe/.report.ErrorActivity t1711}}} to stack=1 task=1711 at 0
11-23 19:03:45.041 D/MountService(2906): getExternalStorageMountMode : final mountMode=3, uid : 10753, packageName : org.schabi.newpipe
11-23 19:03:45.056 W/SELinux (355): SELinux: seapp_context_lookup: seinfo=default, level=s0:c512,c768, pkgname=org.schabi.newpipe 
11-23 19:03:45.061 I/ActivityManager(2906): Start proc 355:org.schabi.newpipe/u0a753 for activity org.schabi.newpipe/.report.ErrorActivity
11-23 19:03:45.076 V/WindowStateAnimator(2906): Finishing drawing window Window{dee9bd5 u0 d0 p2906 Starting org.schabi.newpipe}: mDrawState=DRAW_PENDING
11-23 19:03:45.101 D/ActivityManager(2906):  Launching org.schabi.newpipe, updated priority
11-23 19:03:45.111 D/GameManagerService(2906): NotifyRunnable. pkg: org.schabi.newpipe, type: 4, isMinimized: false, isTunableApp: false
11-23 19:03:45.126 V/MARsPolicyManager(2906): updatePackagesScore PackageInfo name -- org.schabi.newpipe
11-23 19:03:45.131 D/GameManagerService(2906): identifyGamePackage. org.schabi.newpipe
11-23 19:03:45.131 D/StatusBarManagerService(2906): manageDisableList userId=0 what=0x0 pkg=Window{dee9bd5 u0 d0 p2906 Starting org.schabi.newpipe}
11-23 19:03:45.131 D/GameManagerService(2906): identifyGamePackage. org.schabi.newpipe
11-23 19:03:45.131 D/GameManagerService(2906): identifyGamePackage. org.schabi.newpipe
11-23 19:03:45.146 W/System  (355): ClassLoader referenced unknown path: /data/app/org.schabi.newpipe-1/lib/arm
11-23 19:03:45.226 I/ACRA    (355): ACRA is enabled for org.schabi.newpipe, initializing...
11-23 19:03:45.896 E/ACRA    (355): ACRA caught a IllegalAccessError for org.schabi.newpipe
11-23 19:03:45.896 E/ACRA    (355): java.lang.IllegalAccessError: Method 'void org.schabi.newpipe.report.ErrorInfo.<init>(org.schabi.newpipe.report.UserAction, java.lang.String, java.lang.String, int)' is inaccessible to class 'org.schabi.newpipe.report.ErrorInfo$Creator' (declaration of 'org.schabi.newpipe.report.ErrorInfo$Creator' appears in /data/app/org.schabi.newpipe-1/base.apk)
11-23 19:03:45.896 E/ACRA    (355): 	at org.schabi.newpipe.report.ErrorInfo$Creator.createFromParcel(Unknown Source)
11-23 19:03:45.896 E/ACRA    (355): 	at org.schabi.newpipe.report.ErrorActivity.onCreate(ErrorActivity.java:209)
11-23 19:03:46.306 I/WindowManager(2906): Screenshot max retries 4 of Token{c1aa06 ActivityRecord{2493ce1 u0 org.schabi.newpipe/.report.ErrorActivity t1711 f}} appWin=Window{dee9bd5 u0 d0 p2906 Starting org.schabi.newpipe} drawState=4
11-23 19:03:46.321 D/PackageManager(2906): getComponentMetadataForIconTray : org.schabi.newpipe.report.ErrorActivity does not exist in mServices
11-23 19:03:46.321 D/PackageManager(2906): getComponentMetadataForIconTray : org.schabi.newpipe.report.ErrorActivity does not exist in mProviders
11-23 19:03:46.321 D/PackageManager(2906): getComponentMetadataForIconTray : org.schabi.newpipe.report.ErrorActivity does not exist in mReceivers
11-23 19:03:46.321 I/ApplicationPackageManager(3274): load=org.schabi.newpipe, bg=96-96, dr=96-96
11-23 19:03:46.446 I/ActivityManager(2906): Process org.schabi.newpipe (pid 355)(adj 0) has died(96,242)
11-23 19:03:46.446 D/ActivityManager(2906): removeProcessNameLocked mProcessNames.remove pid=355 ,hash=231829482 ,name=org.schabi.newpipe
11-23 19:03:47.421 I/ACRA    (333): Sending report /data/user/0/org.schabi.newpipe/app_ACRA-approved/2020-11-23T19:03:45.939+05:30.stacktrace
11-23 19:03:47.431 I/Timeline(333): Timeline: Activity_launch_request id:org.schabi.newpipe time:325279252
11-23 19:03:47.436 E/Parcel  (2906): Class not found when unmarshalling: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:47.436 E/Parcel  (2906): java.lang.ClassNotFoundException: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:47.436 E/Parcel  (2906): Caused by: java.lang.ClassNotFoundException: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:47.436 I/ActivityManager(2906): isWidgetUsingPkg : org.schabi.newpipe no widget is using.
11-23 19:03:47.446 V/WindowManager(2906): addAppToken: AppWindowToken{14f2445 token=Token{2d94cbc ActivityRecord{17f80af u0 org.schabi.newpipe/.report.ErrorActivity t1712}}} to stack=1 task=1712 at 0
11-23 19:03:47.456 D/MountService(2906): getExternalStorageMountMode : final mountMode=3, uid : 10753, packageName : org.schabi.newpipe
11-23 19:03:47.471 I/ActivityManager(2906): Start proc 404:org.schabi.newpipe/u0a753 for activity org.schabi.newpipe/.report.ErrorActivity
11-23 19:03:47.476 W/SELinux (404): SELinux: seapp_context_lookup: seinfo=default, level=s0:c512,c768, pkgname=org.schabi.newpipe 
11-23 19:03:47.491 V/WindowStateAnimator(2906): Finishing drawing window Window{37a96fd u0 d0 p2906 Starting org.schabi.newpipe}: mDrawState=DRAW_PENDING
11-23 19:03:47.511 D/ActivityManager(2906):  Launching org.schabi.newpipe, updated priority
11-23 19:03:47.521 D/GameManagerService(2906): NotifyRunnable. pkg: org.schabi.newpipe, type: 4, isMinimized: false, isTunableApp: false
11-23 19:03:47.531 V/MARsPolicyManager(2906): updatePackagesScore PackageInfo name -- org.schabi.newpipe
11-23 19:03:47.531 D/StatusBarManagerService(2906): manageDisableList userId=0 what=0x0 pkg=Window{37a96fd u0 d0 p2906 Starting org.schabi.newpipe}
11-23 19:03:47.551 W/System  (404): ClassLoader referenced unknown path: /data/app/org.schabi.newpipe-1/lib/arm
11-23 19:03:47.621 I/ACRA    (404): ACRA is enabled for org.schabi.newpipe, initializing...
11-23 19:03:48.306 E/ACRA    (404): ACRA caught a IllegalAccessError for org.schabi.newpipe
11-23 19:03:48.306 E/ACRA    (404): java.lang.IllegalAccessError: Method 'void org.schabi.newpipe.report.ErrorInfo.<init>(org.schabi.newpipe.report.UserAction, java.lang.String, java.lang.String, int)' is inaccessible to class 'org.schabi.newpipe.report.ErrorInfo$Creator' (declaration of 'org.schabi.newpipe.report.ErrorInfo$Creator' appears in /data/app/org.schabi.newpipe-1/base.apk)
11-23 19:03:48.306 E/ACRA    (404): 	at org.schabi.newpipe.report.ErrorInfo$Creator.createFromParcel(Unknown Source)
11-23 19:03:48.306 E/ACRA    (404): 	at org.schabi.newpipe.report.ErrorActivity.onCreate(ErrorActivity.java:209)
11-23 19:03:48.691 I/WindowManager(2906): Screenshot max retries 4 of Token{2d94cbc ActivityRecord{17f80af u0 org.schabi.newpipe/.report.ErrorActivity t1712 f}} appWin=Window{37a96fd u0 d0 p2906 Starting org.schabi.newpipe} drawState=4
11-23 19:03:48.711 D/PackageManager(2906): getComponentMetadataForIconTray : org.schabi.newpipe.report.ErrorActivity does not exist in mServices
11-23 19:03:48.711 D/PackageManager(2906): getComponentMetadataForIconTray : org.schabi.newpipe.report.ErrorActivity does not exist in mProviders
11-23 19:03:48.711 D/PackageManager(2906): getComponentMetadataForIconTray : org.schabi.newpipe.report.ErrorActivity does not exist in mReceivers
11-23 19:03:48.716 I/ApplicationPackageManager(3274): load=org.schabi.newpipe, bg=96-96, dr=96-96
11-23 19:03:48.821 I/ActivityManager(2906): Process org.schabi.newpipe (pid 404)(adj 1) has died(94,242)
11-23 19:03:48.821 D/ActivityManager(2906): removeProcessNameLocked mProcessNames.remove pid=404 ,hash=165777394 ,name=org.schabi.newpipe
11-23 19:03:49.816 I/ACRA    (333): Sending report /data/user/0/org.schabi.newpipe/app_ACRA-approved/2020-11-23T19:03:48.338+05:30.stacktrace
11-23 19:03:49.826 I/Timeline(333): Timeline: Activity_launch_request id:org.schabi.newpipe time:325281646
11-23 19:03:49.831 E/Parcel  (2906): Class not found when unmarshalling: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:49.831 E/Parcel  (2906): java.lang.ClassNotFoundException: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:49.831 E/Parcel  (2906): Caused by: java.lang.ClassNotFoundException: org.schabi.newpipe.report.ErrorInfo
11-23 19:03:49.831 I/ActivityManager(2906): isWidgetUsingPkg : org.schabi.newpipe no widget is using.
```
</p>
</details>

### Device info

 - Android version/Custom ROM version: Android 6.0.1
 - Device model: 
Samsung SM-G550FY

### PR DESCRIPTION
**Pull request by XiangRongLin**
#### What is it?
- [x] Bugfix (user facing)

#### Fixes the following issue(s)

-  closes #4980

#### APK testing 

[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/5591852/app-debug.zip)
Note that loading comments always throws an error in this apk, since i changed the extractor for easier manual testing

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

@TobiGr @Stypox

### CODE REVIEW

**Comment by okan35** [comment]
To me it still goes into loop of restart when I report.

**Comment by XiangRongLin** [comment]
@okan35 What device/software version are you using. I retried it with the zip from here on my phone and on an Android 4.4 emulator and it shows me the error screen there.

**Comment by okan35** [comment]
I use Sony xperia xz premium, android version is android 9.

**Comment by XiangRongLin** [comment]
@okan35 I retried it with an android 9 emulator and can't reproduce it. Must be something specific to your device maybe

File: app/src/main/java/org/schabi/newpipe/report/ErrorInfo.kt [hunk 1]
```diff
@@ -5,7 +5,7 @@ import androidx.annotation.StringRes
 import kotlinx.android.parcel.Parcelize
 
 @Parcelize
-class ErrorInfo private constructor(
+class ErrorInfo constructor(
```
Comment (Isira-Seneviratne) [inline app/src/main/java/org/schabi/newpipe/report/ErrorInfo.kt:8]: The `constructor` keyword can be removed as well.

Comment (XiangRongLin) [reply app/src/main/java/org/schabi/newpipe/report/ErrorInfo.kt]: good catch

**Comment by FireMasterK** [comment]
Can confirm the issue has been fixed, LGTM

**Comment by TiA4f8R** [comment]
@XiangRongLin I tried your APK and the crash UI is showing for comments. However, because I haven't a default browser, the app is crashing (because of Huawei modifications, this bug is #3925) too but without showing Crash UI Error (it just goes back to home screen) when I press Open in browser button on a video (this is happening on playlists too with the corresponding command). Here is a screencast for reference:

[Crash NewPipe open in browser vs crash comments debug.mp4.zip](https://github.com/TeamNewPipe/NewPipe/files/5599118/Crash.NewPipe.open.in.browser.vs.crash.comments.debug.mp4.zip)

My device is a Honor 9X running EMUI 10.0.0 and Android 10.

**Comment by FireMasterK** [comment]
@TiA4f8R that seems to be a totally different error, this one would take you into a loop of the app opening and closing on it's own.

**Comment by c-shubh** [comment]
Works fine, app doesn't crash.

**Review by TobiGr — APPROVED** [review]
Thank you

**Pull request merged**